torch/profiler/_utils.py

```diff
@@ -40,15 +41,19 @@ def __eq__(self, other):
         return self.event.id == other.event.id
 
     def __repr__(self):
-        return f"<{self.event.name()} id={self.event.correlation_id}>"
+        return f"{self.event.name()}"
 
     def intervals_overlap(self, intervals: List[Interval]):
         overlap_time = 0
         intervals = sorted(intervals, key=lambda x: x.start)
         for i, interval in enumerate(intervals):
-            if i + 1 < len(intervals):
-                assert interval.end <= intervals[
-                    i + 1].start, "Intervals must be disjoint"
+            # Interval overlaps with next interval
+            if i + 1 < len(intervals) and interval.end > intervals[i +
+                                                                   1].start:
+                if interval.end >= intervals[i + 1].end:
+                    del intervals[i + 1]
```

Comment:
Don't mutate a list while you're iterating over it.

Reply:
Change the logic, and added a test to check my sanity